src/contexts/Localization/Provider.tsx

```diff
@@ -0,0 +1,106 @@
+import React, { createContext, useCallback, useEffect, useState } from 'react'
+import { EN, languages, Language } from '../../config/localization/languages'
+import translations from '../../config/localization/translations.json'
+import { ContextApi, ContextData, ProviderState } from './types'
+import { LS_KEY, fetchLocale, getLanguageCodeFromLS } from './helpers'
+
+const initialState: ProviderState = {
+  isFetching: true,
+  currentLanguage: EN,
+}
+
+// Export the translations directly
+export const languageMap = new Map<Language['code'], Record<string, string>>()
+languageMap.set(EN.code, translations)
+
+export const LanguageContext = createContext<ContextApi>(undefined)
+
+export const LanguageProvider: React.FC = ({ children }) => {
+  const [state, setState] = useState<ProviderState>(() => {
+    const codeFromStorage = getLanguageCodeFromLS()
+
+    return {
+      ...initialState,
+      currentLanguage: languages[codeFromStorage],
+    }
+  })
+  const { currentLanguage } = state
+
+  useEffect(() => {
+    const fetchInitialLocales = async () => {
+      const codeFromStorage = getLanguageCodeFromLS()
+
+      if (codeFromStorage !== EN.code) {
+        const enLocale = languageMap.get(EN.code)
+        const currentLocale = await fetchLocale(codeFromStorage)
```

Comment:
Is a 'locale' here describing a json of all translations for that specific language?